rpc/kate-rpc/src/lib.rs

```diff
@@ -454,3 +480,12 @@ fn non_extended_dimensions(ext_data: &DMatrix<BlsScalar>) -> RpcResult<Dimension
 
 	Ok(dimensions)
 }
+
+fn is_da_transaction<A, C, D: SignedExtension>(
```

Comment:
That is NOT going to work if a Runtime Upgrade affects the DA calls. You should re-use the filters we have in Runtime, and call it through the runtime api, like `self.client.runtime_api().<filter_function>( raw_extrinsic)`

Reply:
wdyt of using `transaction_index` in `submitted_data::calls_proof` instead of leaf index, and compute leaf index inside?

Reply:
I think we should move completely the Kate RPC into LC logic or/and make some utilities in `avail-subxt`. I mean there is **no needed to run this code inside a node**. att @aterentic-ethernal

Reply:
If I'm not missing something, this has to be run by the node since LC doesn't fetch all transactions, only transactions related to specified application, which is not enough to generate proof.